service-monitor/src/main/java/com/yahoo/vespa/service/health/ApacheHttpClient.java

```diff
@@ -57,16 +54,23 @@ public long getKeepAliveDuration(HttpResponse response, HttpContext context) {
                     }
                 };
 
-        return HttpClients.custom()
+        return VespaHttpClientBuilder.create(ApacheHttpClient::createConnectionManager)
                 .setKeepAliveStrategy(keepAliveStrategy)
-                .setConnectionManager(connectionManager)
                 .disableAutomaticRetries()
                 .setDefaultRequestConfig(requestConfig)
                 .build();
     }
 
-    ApacheHttpClient(URL url, Duration timeout, Duration keepAlive, ConnectionSocketFactory socketFactory) {
-        this(url, makeCloseableHttpClient(url, timeout, keepAlive, socketFactory));
+    private static HttpClientConnectionManager createConnectionManager(SSLConnectionSocketFactory sslConnectionSocketFactory) {
```

Comment:
I guess this would be ideal in VespaHttpClientBuilder if other clients wants the same behavior.

Reply:
I have added a new static factory method that does that. See new commits.